services/src/test/scala/cromwell/services/metadata/impl/MetadataDatabaseAccessSpec.scala

```diff
@@ -473,9 +473,44 @@ class MetadataDatabaseAccessSpec extends FlatSpec with Matchers with ScalaFuture
       } yield()).futureValue
     }
 
+    it should "properly query metadata summaries based on archived status and timestamp and update archive status after metadata deletion" taggedAs DbmsTest in {
+      val defaultTimeout: FiniteDuration = 60.seconds
+
+      val workflowId1Future = succeededWorkflowMetadata(WorkflowId.fromString("11111111-1111-1111-1111-111111111112"))
+      val workflowId2Future = succeededWorkflowMetadata(WorkflowId.fromString("11111111-1111-1111-1111-111111111113"))
+      val workflowId1 = Await.result(workflowId1Future, defaultTimeout)
+      val workflowId2 = Await.result(workflowId2Future, defaultTimeout)
+
+      val refreshSummariesFuture = dataAccess.refreshWorkflowMetadataSummaries(1000, Option(Long.MaxValue))
+      Await.result(refreshSummariesFuture, defaultTimeout)
+
+      val updateMetadataArchiveStatusFuture = dataAccess.updateMetadataArchiveStatus(workflowId1, MetadataArchiveStatus.Archived)
+      Await.result(updateMetadataArchiveStatusFuture, defaultTimeout)
+
+      val responseEmptyFuture = dataAccess.queryRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp(Option("Archived"), OffsetDateTime.now().minusMinutes(1), 200)
+      val responseEmpty = Await.result(responseEmptyFuture, defaultTimeout)
+      responseEmpty shouldBe empty
+
+      eventually(Timeout(defaultTimeout), Interval(2.seconds)) {
+        val responseNonEmptyFuture = dataAccess.queryRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp (Option ("Archived"), OffsetDateTime.now().minusMinutes(1), 200)
+        val responseNonEmpty = Await.result(responseNonEmptyFuture, 10.seconds)
```

Comment:
is this `eventually` + `Await` doing what you want? 

It looks like it will keep trying the query until it is able to get a response within 10 seconds (up to a total timeout of 60 seconds)

Reply:
I expect it to always be able to get response from the query within 10 seconds. The question here is content of the response - eventually it should become not empty.
So it's more like "within 60 seconds make the query with 2-second intervals until this query returns non-empty response".
But I'm going to adjust the timings a bit.

Reply:
Oh, I see. Got it. It's the response that's being asserted to eventually look good. In that case 👍